resources/sdk/purecloudgo/templates/logger.mustache

```diff
@@ -0,0 +1,248 @@
+	RequestHeaders  http.Header `json:"requestHeaders,omitempty"`
+	ResponseHeaders http.Header `json:"responseHeaders,omitempty"`
+	CorrelationId   string      `json:"correlationId,omitempty"`
+	StatusCode      int         `json:"statusCode,omitempty"`
+	RequestBody     string      `json:"requestBody,omitempty"`
+	ResponseBody    string      `json:"responseBody,omitempty"`
+}
+
+func (s *logStatement) string(format LoggingFormat, logRequestBody, logResponseBody bool) string {
+	if len(s.RequestHeaders["Authorization"]) > 0 {
+		s.RequestHeaders["Authorization"] = []string{"[REDACTED]"}
+	}
+
+	if !logRequestBody {
+		s.RequestBody = ""
+	}
+	if !logResponseBody {
+		s.ResponseBody = ""
+	}
+
+	if format == Text {
+		return fmt.Sprintf(`
+=== REQUEST ===%v%v%v%v
+=== RESPONSE ===%v%v%v%v`,
+			formatValue("URL", s.URL),
+			formatValue("Method", s.Method),
+			formatValue("Headers", formatHeaders(s.RequestHeaders)),
+			formatValue("Body", s.RequestBody),
+
+			formatValue("Status", fmt.Sprintf("%v", s.StatusCode)),
+			formatValue("Headers", formatHeaders(s.ResponseHeaders)),
+			formatValue("CorrelationId", s.CorrelationId),
+			formatValue("Body", s.ResponseBody))
+	}
+
+	j, _ := json.Marshal(s)
+	str, _ := strconv.Unquote(strings.Replace(strconv.Quote(string(j)), `\\u`, `\u`, -1))
+
+	return strings.TrimRight(string(pretty.Ugly([]byte(str))), "\n")
+}
+
+func (c *LoggingConfiguration) configureLogging() {
+	var f *os.File
+	if c.logFilePath != "" {
+		f, _ = os.OpenFile(c.logFilePath, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0666)
+	}
+
+	var stdoutWrt io.Writer
+	var stderrWrt io.Writer
+
+	if f != nil && c.logToConsole { // Logging to console and file
+		stdoutWrt = io.MultiWriter(f, os.Stdout)
+		stderrWrt = io.MultiWriter(f, os.Stderr)
+	} else if f == nil && c.logToConsole { // Logging to console
+		stdoutWrt = io.MultiWriter(os.Stdout)
+		stderrWrt = io.MultiWriter(os.Stderr)
+	} else if f != nil && !c.logToConsole { // Logging to file
+		stdoutWrt = io.MultiWriter(f)
+		stderrWrt = io.MultiWriter(f)
+	} else { // Cannot log to anything
+		traceLogger = nil
+		debugLogger = nil
+		errorLogger = nil
+		return
+	}
+
+	flags := 0
+	tracePrefix := ""
+	debugPrefix := ""
+	errorPrefix := ""
+	if c.logFormat == Text {
+		flags = log.Ldate | log.Ltime
+		tracePrefix = "TRACE: "
+		debugPrefix = "DEBUG: "
+		errorPrefix = "ERROR: "
+	}
+
+	traceLogger = log.New(stdoutWrt, tracePrefix, flags)
+	debugLogger = log.New(stdoutWrt, debugPrefix, flags)
+	errorLogger = log.New(stderrWrt, errorPrefix, flags)
+}
+
+func (c *LoggingConfiguration) trace(method, URL string, requestBody []byte, statusCode int, requestHeaders, responseHeaders http.Header) {
+	now := time.Now()
+	logStatement := &logStatement{
+		Date:            &now,
+		Level:           "trace",
+		Method:          method,
+		URL:             URL,
+		RequestBody:     string(requestBody),
+		StatusCode:      statusCode,
+		RequestHeaders:  requestHeaders,
+		ResponseHeaders: responseHeaders,
+	}
```

Comment:
According to the design doc, I think trace level log should also contain a correlation Id?

Reply:
You're right, thanks for pointing that out